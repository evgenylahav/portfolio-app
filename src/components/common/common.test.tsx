import React from 'react';
import ReactDOM, { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { BlogPost, CodeSnippet, Skills, Social, ResumeLine, Jumbotron, FooterComp, NavigationBar } from '.';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { testNameToKey } from 'jest-snapshot/build/utils';

const renderTester = (component) => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
};

const shallowTester = (component, tag, value) => {
    const element = shallow(component);
    return expect(element.find(`[data-test='${tag}']`).length).toBe(value);
}

const testSnapShot = (inComponent) => {
    const component = renderer.create(inComponent);
    const tree = component.toJSON();
    return expect(tree).toMatchSnapshot();
}

describe('Footer Component unit test', () => {
    it('renders without crashing', () => {
      renderTester(<FooterComp />)
    });
  
    it('checks that only one footer component is created', () => {
      shallowTester(<FooterComp />, 'footer-row', 1);
    });
  
    test('has a valid snapshot', () => {  
      testSnapShot(<FooterComp />);
    });
});


describe('Jumbotron unit test', () => {
    it('renders without crashing', () => {
      renderTester(<Jumbotron />)
    });
  
    it('checks that only one jumbotron is created', () => {
      shallowTester(<Jumbotron />, 'jumbotron-test', 1);
    });
  
    test('has a valid snapshot', () => {  
      testSnapShot(<Jumbotron />);
    });
});


describe('Resume line unit test', () => {
    const props = {
        img: 1,
        startYear: '1955',
        endYear: '1978',
        placeName: 'Aquarium',
        position: 'shark eater',
        description: ['eating', 'many', 'sharks']
    };
    it('renders without crashing', () => {
        renderTester(<ResumeLine {...props} />)
    });
  
    it('checks that only one skill is created', () => {
        shallowTester(<ResumeLine {...props} />, 'resume-line-container', 1);
    });
  
    test('has a valid snapshot', () => {
        testSnapShot(<ResumeLine {...props} />);
    });
  });


describe('Social unit test', () => {
    it('renders without crashing', () => {
      renderTester(<Social />)
    });
  
    it('checks that only one skill is created', () => {
      shallowTester(<Social />, 'social-div', 1);
    });
  
    test('has a valid snapshot', () => {  
      testSnapShot(<Social />);
    });
});


describe('Skills unit test', () => {
    const props = {
        skill: 'running',
        skillLevel: 55,
    };
    it('renders without crashing', () => {
        renderTester(<Skills {...props} />)
    });
  
    it('checks that only one skill is created', () => {
        shallowTester(<Skills {...props} />, 'skills-container', 1);
    });
  
    test('has a valid snapshot', () => {
        testSnapShot(<Skills {...props} />);
    });
  });


describe('Code snippet unit test', () => {
    const props = {
      code: 'import hello',
      language: 'python',
    };
    it('renders without crashing', () => {
        renderTester(<CodeSnippet {...props} />)
    });
  
    it('checks that only one blog post is created', () => {
        shallowTester(<CodeSnippet {...props} />, 'code-snippet-div', 1);
    });
  
    // test('has a valid snapshot', () => {
    //     testSnapShot(<CodeSnippet {...props} />);
    // });
  });
  


describe('Blog post unit test', () => {
  const props = {
    title: "blog title",
    header: "blog header",
    text: "bla bla bla",
    url: "http://www.google.com/",
    date: "23-March-2019",
    tags: "python",
    img: "temp"
  };
  it('renders without crashing', () => {
    renderTester(
        <MemoryRouter>
            <BlogPost {...props} />
        </MemoryRouter>
    );
  });

  it('checks that only one blog post is created', () => {
    shallowTester(<BlogPost {...props} />, 'blog-card', 1);
  });

  test('has a valid snapshot', () => {
    testSnapShot(
        <MemoryRouter>
            <BlogPost {...props} />
        </MemoryRouter>
    );
  });
});

describe('Navigation bar unit test', () => {
    it('renders without crashing', () => {
      renderTester(
        <MemoryRouter>
          <NavigationBar />
        </MemoryRouter>
      );
    });
  
    it('checks that only one navbar is created', () => {
      shallowTester(<NavigationBar />, 'navbar-styles', 1);
    });
  
    test('has a valid snapshot', () => {  
      testSnapShot(
        <MemoryRouter>
            <NavigationBar />
        </MemoryRouter>
      );
    });
});