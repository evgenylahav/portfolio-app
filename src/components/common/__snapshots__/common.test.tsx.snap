// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component unit test has a valid snapshot 1`] = `
<div
  className="row"
  data-test="footer-row"
  style={
    Object {
      "backgroundColor": "#aaa",
    }
  }
>
  <div
    className="col-sm-4 col-12"
    style={
      Object {
        "margin": "30px",
      }
    }
  >
    <span
      style={
        Object {
          "margin": "50px",
        }
      }
    >
        Evgeny Lahav
    </span>
  </div>
  <div
    className="col-sm-6 col-12"
    style={
      Object {
        "margin": "20px",
        "marginLeft": "50px",
      }
    }
  >
    <div
      data-test="social-div"
    >
      <div
        className="social-links row"
      >
        <a
          className="aws-btn aws-btn--twitter aws-btn--visible"
          href="https://twitter.com/kodon23"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          role="button"
          style={Object {}}
          target="_blank"
          title={null}
          to={null}
        >
          <span
            className="aws-btn__wrapper"
          >
            <span
              className="aws-btn__content"
            >
              <span>
                <svg
                  height={23}
                  preserveAspectRatio="xMidYMid meet"
                  space="preserve"
                  style={
                    Object {
                      "marginRight": undefined,
                    }
                  }
                  version="1.1"
                  viewBox="0 0 50 50"
                  width={30}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M49.3,9.1c-1.8,0.8-3.6,1.3-5.6,1.5c2-1.2,3.6-3.1,4.3-5.4c-1.9,1.1-4,1.9-6.2,2.4c-1.8-1.9-4.3-3.1-7.1-3.1c-5.4,0-9.8,4.4-9.8,9.8c0,0.8,0.1,1.5,0.3,2.2C17,16.1,9.8,12.2,5,6.3c-0.8,1.4-1.3,3.1-1.3,4.9c0,3.4,1.7,6.4,4.3,8.1c-1.6-0.1-3.1-0.5-4.4-1.2v0.1c0,4.7,3.4,8.7,7.8,9.6c-0.8,0.2-1.7,0.3-2.6,0.3c-0.6,0-1.2-0.1-1.8-0.2c1.2,3.9,4.9,6.7,9.1,6.8c-3.3,2.6-7.6,4.2-12.1,4.2c-0.8,0-1.6,0-2.3-0.1c4.3,2.8,9.5,4.4,15,4.4c18,0,27.8-14.9,27.8-27.8c0-0.4,0-0.8,0-1.3C46.4,12.7,48,11,49.3,9.1z"
                    fill="#FFF"
                  />
                </svg>
                 
              </span>
            </span>
          </span>
        </a>
        <a
          className="aws-btn aws-btn--github aws-btn--visible"
          href="https://github.com/evgenylahav"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          role="button"
          style={Object {}}
          target="_blank"
          title={null}
          to={null}
        >
          <span
            className="aws-btn__wrapper"
          >
            <span
              className="aws-btn__content"
            >
              <span>
                <svg
                  height={23}
                  preserveAspectRatio="xMidYMid meet"
                  space="preserve"
                  style={
                    Object {
                      "marginRight": undefined,
                    }
                  }
                  version="1.1"
                  viewBox="0 0 50 50"
                  width={30}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25,2.5C12.2,2.5,1.9,12.8,1.9,25.6c0,10.2,6.6,18.9,15.8,21.9c1.2,0.2,1.5-0.5,1.5-1.1c0-0.5,0-2.1,0-4c-6.4,1.4-7.8-3-7.8-3c-1-2.7-2.6-3.4-2.6-3.4C6.7,34.6,9,34.6,9,34.6c2.3,0.2,3.5,2.4,3.5,2.4c2.1,3.5,5.4,2.5,6.7,1.9c0.2-1.5,0.8-2.5,1.5-3.1c-5.1-0.6-10.5-2.6-10.5-11.4c0-2.5,0.9-4.6,2.4-6.2c-0.2-0.6-1-2.9,0.2-6.1c0,0,1.9-0.6,6.3,2.4c1.8-0.5,3.8-0.8,5.8-0.8s3.9,0.3,5.8,0.8c4.4-3,6.3-2.4,6.3-2.4c1.3,3.2,0.5,5.5,0.2,6.1c1.5,1.6,2.4,3.7,2.4,6.2c0,8.9-5.4,10.8-10.5,11.4c0.8,0.7,1.6,2.1,1.6,4.3c0,2.9,0,5.6,0,6.4c0,0.6,0.4,1.3,1.5,1.1C41.4,44.5,48,35.9,48,25.7C48.1,12.8,37.8,2.5,25,2.5z"
                    fill="#FFF"
                  />
                </svg>
                 
              </span>
            </span>
          </span>
        </a>
        <a
          className="aws-btn aws-btn--linkedin aws-btn--visible"
          href="https://www.linkedin.com/in/evgeny-lahav"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          role="button"
          style={Object {}}
          target="_blank"
          title={null}
          to={null}
        >
          <span
            className="aws-btn__wrapper"
          >
            <span
              className="aws-btn__content"
            >
              <span>
                <svg
                  height={23}
                  preserveAspectRatio="xMidYMid meet"
                  space="preserve"
                  style={
                    Object {
                      "marginRight": undefined,
                    }
                  }
                  version="1.1"
                  viewBox="0 0 50 50"
                  width={30}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.8,41.7H5.4V13.6h8.4C13.8,13.6,13.8,41.7,13.8,41.7z M44.7,41.7h-8.4v-15c0-3.9-1.4-5.9-4.2-5.9c-2.2,0-3.6,1.1-4.3,3.3c0,3.5,0,17.6,0,17.6h-8.4c0,0,0.1-25.3,0-28.1h6.7l0.5,5.6h0.2c1.7-2.8,4.5-4.7,8.3-4.7c2.9,0,5.2,0.8,7,2.8s2.7,4.7,2.7,8.5L44.7,41.7L44.7,41.7z M9.6,2.3C12,2.3,14,4.2,14,6.5s-2,4.2-4.4,4.2S5.2,8.8,5.2,6.5C5.3,4.2,7.2,2.3,9.6,2.3z"
                    fill="#FFF"
                  />
                </svg>
                 
              </span>
            </span>
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Jumbotron unit test has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa lmuQjA"
  data-test="jumbotron-test"
>
  <div
    className="jumbo jumbotron jumbotron-fluid"
  >
    <div
      className="overlay"
    />
    <div
      className="container"
    >
      <h1>
        Welcome to my page
      </h1>
    </div>
  </div>
</div>
`;

exports[`Resume line unit test has a valid snapshot 1`] = `
<div
  className="container"
  data-test="resume-line-container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-12"
    >
      <p
        style={
          Object {
            "fontSize": "18px",
            "paddingTop": "5px",
          }
        }
      >
        1955
         - 
        1978
      </p>
    </div>
    <div
      className="col-sm-9 col-12"
    >
      <img
        className="resume-place-name-image img-thumbnail"
        src={1}
      />
      <h4
        style={
          Object {
            "marginTop": "0px",
          }
        }
      >
        Aquarium
      </h4>
      <h6
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        shark eater
      </h6>
      <ul>
         
        <li>
          eating
        </li>
        <li>
          many
        </li>
        <li>
          sharks
        </li>
         
      </ul>
    </div>
  </div>
</div>
`;

exports[`Skills unit test has a valid snapshot 1`] = `
<div
  className="container"
  data-test="skills-container"
>
  <div
    className="col-sm-8 col-12"
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      running
    </div>
  </div>
  <div
    className="col-sm-12 col-12"
  >
    <div>
      <div
        className="progress"
        style={
          Object {
            "color": "green",
            "marginBottom": "20px",
          }
        }
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={55}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "55%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Social unit test has a valid snapshot 1`] = `
<div
  data-test="social-div"
>
  <div
    className="social-links row"
  >
    <a
      className="aws-btn aws-btn--twitter aws-btn--visible"
      href="https://twitter.com/kodon23"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      role="button"
      style={Object {}}
      target="_blank"
      title={null}
      to={null}
    >
      <span
        className="aws-btn__wrapper"
      >
        <span
          className="aws-btn__content"
        >
          <span>
            <svg
              height={23}
              preserveAspectRatio="xMidYMid meet"
              space="preserve"
              style={
                Object {
                  "marginRight": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 50 50"
              width={30}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M49.3,9.1c-1.8,0.8-3.6,1.3-5.6,1.5c2-1.2,3.6-3.1,4.3-5.4c-1.9,1.1-4,1.9-6.2,2.4c-1.8-1.9-4.3-3.1-7.1-3.1c-5.4,0-9.8,4.4-9.8,9.8c0,0.8,0.1,1.5,0.3,2.2C17,16.1,9.8,12.2,5,6.3c-0.8,1.4-1.3,3.1-1.3,4.9c0,3.4,1.7,6.4,4.3,8.1c-1.6-0.1-3.1-0.5-4.4-1.2v0.1c0,4.7,3.4,8.7,7.8,9.6c-0.8,0.2-1.7,0.3-2.6,0.3c-0.6,0-1.2-0.1-1.8-0.2c1.2,3.9,4.9,6.7,9.1,6.8c-3.3,2.6-7.6,4.2-12.1,4.2c-0.8,0-1.6,0-2.3-0.1c4.3,2.8,9.5,4.4,15,4.4c18,0,27.8-14.9,27.8-27.8c0-0.4,0-0.8,0-1.3C46.4,12.7,48,11,49.3,9.1z"
                fill="#FFF"
              />
            </svg>
             
          </span>
        </span>
      </span>
    </a>
    <a
      className="aws-btn aws-btn--github aws-btn--visible"
      href="https://github.com/evgenylahav"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      role="button"
      style={Object {}}
      target="_blank"
      title={null}
      to={null}
    >
      <span
        className="aws-btn__wrapper"
      >
        <span
          className="aws-btn__content"
        >
          <span>
            <svg
              height={23}
              preserveAspectRatio="xMidYMid meet"
              space="preserve"
              style={
                Object {
                  "marginRight": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 50 50"
              width={30}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25,2.5C12.2,2.5,1.9,12.8,1.9,25.6c0,10.2,6.6,18.9,15.8,21.9c1.2,0.2,1.5-0.5,1.5-1.1c0-0.5,0-2.1,0-4c-6.4,1.4-7.8-3-7.8-3c-1-2.7-2.6-3.4-2.6-3.4C6.7,34.6,9,34.6,9,34.6c2.3,0.2,3.5,2.4,3.5,2.4c2.1,3.5,5.4,2.5,6.7,1.9c0.2-1.5,0.8-2.5,1.5-3.1c-5.1-0.6-10.5-2.6-10.5-11.4c0-2.5,0.9-4.6,2.4-6.2c-0.2-0.6-1-2.9,0.2-6.1c0,0,1.9-0.6,6.3,2.4c1.8-0.5,3.8-0.8,5.8-0.8s3.9,0.3,5.8,0.8c4.4-3,6.3-2.4,6.3-2.4c1.3,3.2,0.5,5.5,0.2,6.1c1.5,1.6,2.4,3.7,2.4,6.2c0,8.9-5.4,10.8-10.5,11.4c0.8,0.7,1.6,2.1,1.6,4.3c0,2.9,0,5.6,0,6.4c0,0.6,0.4,1.3,1.5,1.1C41.4,44.5,48,35.9,48,25.7C48.1,12.8,37.8,2.5,25,2.5z"
                fill="#FFF"
              />
            </svg>
             
          </span>
        </span>
      </span>
    </a>
    <a
      className="aws-btn aws-btn--linkedin aws-btn--visible"
      href="https://www.linkedin.com/in/evgeny-lahav"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      role="button"
      style={Object {}}
      target="_blank"
      title={null}
      to={null}
    >
      <span
        className="aws-btn__wrapper"
      >
        <span
          className="aws-btn__content"
        >
          <span>
            <svg
              height={23}
              preserveAspectRatio="xMidYMid meet"
              space="preserve"
              style={
                Object {
                  "marginRight": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 50 50"
              width={30}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.8,41.7H5.4V13.6h8.4C13.8,13.6,13.8,41.7,13.8,41.7z M44.7,41.7h-8.4v-15c0-3.9-1.4-5.9-4.2-5.9c-2.2,0-3.6,1.1-4.3,3.3c0,3.5,0,17.6,0,17.6h-8.4c0,0,0.1-25.3,0-28.1h6.7l0.5,5.6h0.2c1.7-2.8,4.5-4.7,8.3-4.7c2.9,0,5.2,0.8,7,2.8s2.7,4.7,2.7,8.5L44.7,41.7L44.7,41.7z M9.6,2.3C12,2.3,14,4.2,14,6.5s-2,4.2-4.4,4.2S5.2,8.8,5.2,6.5C5.3,4.2,7.2,2.3,9.6,2.3z"
                fill="#FFF"
              />
            </svg>
             
          </span>
        </span>
      </span>
    </a>
  </div>
</div>
`;
